device/quanta/x86_64-quanta_ix1b_rglbmc-r0/plugins/psuutil.py

```diff
@@ -40,12 +42,18 @@ def my_log(txt):
     return
 
 
-def log_os_system(cmd, show):
+def log_os_system(cmd1_args, cmd2_args, show):
+    cmd = ' '.join(cmd1_args) + ' | ' + ' '.join(cmd2_args)
     logging.info('Run :'+cmd)
     status = 1
     output = ""
     try:
-        output = subprocess.check_output(cmd, shell=True, universal_newlines=True)
+        p1 = subprocess.Popen(cmd1_args, universal_newlines=True, stdout=subprocess.PIPE)
+        p2 = subprocess.Popen(cmd2_args, universal_newlines=True, stdin=p1.stdout, stdout=subprocess.PIPE)
+        p1.stdout.close()
```

Comment:
>p1.stdout.close() [](http://example.com/codeflow?start=8&length=17)

If you use `with` on p1 and p2, you do not need explicit `close`? And the benefit is exception-safe.

Reply:
Agree. I updated to use `with`.